librtasevent_src/rtas_srcfru.c

```diff
@@ -73,13 +73,16 @@ parse_fru_id_scn(struct rtas_event *re)
     parse_fru_hdr(&fru_id->fruhdr, &fru_id_raw->fruhdr);
     re->offset += RE_FRU_HDR_SZ;
 
+    int len_procedure_id = strlen(fru_id->procedure_id);
+    int len_part_no = strlen(fru_id->part_no);
+
     if (fruid_has_part_no(fru_id)) {
-        strcpy(fru_id->part_no, RE_EVENT_OFFSET(re));
+        strncpy(fru_id->part_no, RE_EVENT_OFFSET(re), len_part_no);
```

Comment:
why not just call `rtas_copy` function ?

Reply:
thanks for your advise. i overlooked  rtas_copy function :)
Next try:

```
diff -up librtas-2.0.2/librtasevent_src/rtas_srcfru.c.me librtas-2.0.2/librtasevent_src/rtas_srcfru.c
--- librtas-2.0.2/librtasevent_src/rtas_srcfru.c.me     2021-03-26 08:52:29.399602640 +0100
+++ librtas-2.0.2/librtasevent_src/rtas_srcfru.c        2021-03-26 08:52:38.501720153 +0100
@@ -74,13 +74,13 @@ parse_fru_id_scn(struct rtas_event *re)
     re->offset += RE_FRU_HDR_SZ;
 
     if (fruid_has_part_no(fru_id)) {
-        strcpy(fru_id->part_no, RE_EVENT_OFFSET(re));
-        re->offset += 8;
+        rtas_copy(fru_id->part_no, re, 7);
+        fru_id->part_no[7] = '\0';
     }
 
     if (fruid_has_proc_id(fru_id)) {
-        strcpy(fru_id->procedure_id, RE_EVENT_OFFSET(re));
-        re->offset += 8;
+        rtas_copy(fru_id->procedure_id, re, 7);
+        fru_id->procedure_id[7] = '\0';
     }
 
     if (fruid_has_ccin(fru_id)) {
```